### DISCUSSION TITLE
there is a node_modules folder under the .pnpm folder, What is the purpose of this node_modules

### DISCUSSION THREAD

**Discussion by smallwebbird** [pnpm/pnpm#4987] [Q&A] [answered]
there is a node_modules folder under the .pnpm folder, What is the purpose of this node_modules

**Comment by Jack-Works** [comment]
This is where modules actually installed to. See https://pnpm.io/symlinked-node-modules-structure

**Comment by zkochan** [comment, marked as answer]
Some dependencies are broken, they use packages that are not in their dependencies, so we create a hoisted node_modules at `node_modules/.pnpm/node_modules`. You can control what package are hoisted there with the `hoist-pettern` setting.

**Reply by miku03090831** [reply]
Hello, could you consider adding this clarification to the [Symlinked node_modules structure](https://pnpm.io/symlinked-node-modules-structure) section？
After reading this section, I checked my project and found that it didn't align with what was described in the article. According to the article’s node_modules structure, one npm package should not be able to find another package.

It took me some time to realize that this discrepancy was due to the default value of hoist being true. Could you please mention the default value of hoist and its function in the [Symlinked node_modules structure](https://pnpm.io/symlinked-node-modules-structure) article?

**Reply by zkochan** [reply]
ok, updated the docs https://github.com/pnpm/pnpm.io/commit/180c581dc1bd14cbf31658033aa1e3f409dd7b02

**Comment by oneyoung19** [comment]
1、由于历史原因以及 `npm` 的扁平化模块，`npm` 当中可能存在一些包，没有严格遵守“仅引入已经在 `package.json` 中声明的依赖” 这条规则。也就是说，这些历史包可能直接引入 `node_modules` 中的部分依赖，但这部分依赖并没有在 `package.json` 中声明，可以理解为**幻影依赖**。
2、`pnpm` 为了向下兼容这部分包，保证它们的正常运行或者低成本迁移，因而支持配置[hoist](https://pnpm.io/zh/npmrc#hoist)属性以创建 `node_modules/.pnpm/node_modules`。此外，`hoist` 配置默认会提升所有的子依赖到 `node_modules/.pnpm/node_modules` 中，您可以结合 [hoist-pattern](https://pnpm.io/zh/npmrc#hoist-pattern) 来指定可能存在幻影依赖的历史包从而缩小范围。
